testing/unittests/metrics/MetricFrameworkTests.cpp

```diff
@@ -348,18 +348,68 @@ class MetricFrameworkTests : public CppUnit::TestFixture
         pHistogram->recordMeasurement(8);
         sumMeasurements += 8;
         checkHistogramBucketResult(pHistogram, sumMeasurements, {0,2,1,1});
+
+
+        //
+        // Test the scaled time histogram. Buckets for 500ms, 1000ms and 2000ms
+        std::vector<BucketDef> timeBuckets = {{"le 500", 500}, {"le 1000", 1000}, {"le 2000", 2000} };
+        std::shared_ptr<ScaledTimeFromCyclesHistogram> pTimeHistogram = std::make_shared<ScaledTimeFromCyclesHistogram>("requests.time", "description", TimeUnits::HISTOGRAM_TIMEUNITS_MS, timeBuckets);
+        int totalDelay = 0;
+
+        //
+        // test a duration < 500ms
+        {
+            ExecutionTimeHistogramCollector h(pTimeHistogram);
+            usleep(250*1000);
+            totalDelay += 250;
+
+        }
+        checkHistogramBucketResult(pTimeHistogram, -1, {1,0,0,0});
+        CPPUNIT_ASSERT_GREATEREQUAL(totalDelay, static_cast<int>(pTimeHistogram->queryValue()));
+
+        //
+        // duration > 500, < 1000
+        {
+            ExecutionTimeHistogramCollector h(pTimeHistogram);
+            usleep(750*1000);
+            totalDelay += 750;
+        }
+        checkHistogramBucketResult(pTimeHistogram, -1, {1,1,0,0});
+        CPPUNIT_ASSERT_GREATEREQUAL(totalDelay, static_cast<int>(pTimeHistogram->queryValue()));
+
+        //
+        // duration > 1000, 2000
+        {
+            ExecutionTimeHistogramCollector h(pTimeHistogram);
+            usleep(1250*1000);
+            totalDelay += 1250;
+        }
+        checkHistogramBucketResult(pTimeHistogram, -1, {1,1,1,0});
+        CPPUNIT_ASSERT_GREATEREQUAL(totalDelay, static_cast<int>(pTimeHistogram->queryValue()));
+
+        //
+        // duration > 2000
+        {
+            ExecutionTimeHistogramCollector h(pTimeHistogram);
+            usleep(2500*1000);
+            totalDelay += 2500;
+        }
+        checkHistogramBucketResult(pTimeHistogram, -1, {1,1,1,1});
+        CPPUNIT_ASSERT_GREATEREQUAL(totalDelay, static_cast<int>(pTimeHistogram->queryValue()));
     }
 
 
     void checkHistogramBucketResult(std::shared_ptr<IMetric> pHistogram, int expectedSum, const std::vector<int> &expectedValues)
     {
         std::vector<__uint64> values = pHistogram->queryHistogramValues();
-        CPPUNIT_ASSERT_EQUAL(static_cast<int>(pHistogram->queryValue()), expectedSum);
         auto size = expectedValues.size();
         for (unsigned i=0; i<size; ++i)
         {
             CPPUNIT_ASSERT_EQUAL(static_cast<int>(values[i]), expectedValues[i]);
         }
+
+        if (expectedSum > 0)
```

Comment:
trivial: >= 0 would be better - I scanned the test cases for 0s and didn't see any.

Reply:
Agreed. Changed just in case a 0 test was added.